operator/v1/types_csi_cluster_driver.go

```diff
@@ -369,6 +369,21 @@ type VSphereCSIDriverConfigSpec struct {
 	// +openshift:enable:FeatureGate=VSphereDriverConfiguration
 	// +optional
 	GranularMaxSnapshotsPerBlockVolumeInVVOL *uint32 `json:"granularMaxSnapshotsPerBlockVolumeInVVOL,omitempty"`
+
+	// maxAllowedBlockVolumesPerNode is an optional configuration parameter that allows setting custom value for limit of
+	// PersistentVolumes attached to a node. In vSphere version 7 this limit was set to 59 by default, however in
+	// vSphere version 8 this limit was increased to 255. For more details see: https://configmax.broadcom.com/guest?vmwareproduct=vSphere&release=vSphere%208.0&categories=3-0
+	// Before increasing this value above 59 the cluster administrator needs to ensure that every node forming the
+	// cluster is updated to ESXi version 8 or higher and that all nodes are running the same version.
+	// Minimum allowed value: 1
+	// Maximum allowed value: 255 (matches the vSphere version 8 limit)
+	// Default: 59 (matches the vSphere version 7 limit)
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=255
+	// +kubebuilder:default=59
+	// +openshift:enable:FeatureGate=VSphereConfigurableMaxAllowedBlockVolumesPerNode
+	// +optional
+	MaxAllowedBlockVolumesPerNode int32 `json:"maxAllowedBlockVolumesPerNode,omitempty"`
```

Comment:
Is this API likely to be rendered for the end user, or is it something they might craft themselves? Would it improve discoverability if we allowed the 0 value and dropped the omitempty, defining that the 0 value means no opinion.

Reply:
Problem is - `0` basically means there is no limit and interacts weird with rest of the system, unless we changed `0` to mean, OCP will select the default for you.

Reply:
Also possible, but for similar reasons (clarity) as I mentioned in the comment above: in the context of kubernetes node installer manager, scheduler, CSI driver...the value `0` and unset/nil mean "no limit". So making `0` and unset/nil here mean something else (59, or just about anything else than `0`) seems very confusing - perhaps not now but in the future for sure.